lib/jekyll-redirect-from/redirect_page.rb

```diff
@@ -1,42 +1,70 @@
 # Encoding: utf-8
-
 module JekyllRedirectFrom
+  # Specialty page which implements the redirect path logic
   class RedirectPage < Jekyll::Page
-    # Initialize a new RedirectPage.
+    # Use Jekyll's native absolute_url filter
+    include Jekyll::Filters::URLFilters
```

Comment:
I _really_ wish there were a better way to do this.

Maybe the relative URL logic should be broken into a utility method?